### PR TITLE
MinReplicas should not be less than 3

### PR DESCRIPTION
**Pull request by deefreak**

### CODE REVIEW

File: pkg/reco/workflow.go [hunk 1]
```diff
@@ -91,6 +91,7 @@ func (rw *RecommendationWorkflowImpl) Execute(ctx context.Context, wm WorkloadMe
 		rw.logger.Error(err, "Error while generating recommendation")
 		return nil, nil, nil, errors.New("Unable to generate recommendation")
 	}
+	targetRecoConfig = transformTargetRecoConfig(targetRecoConfig)
```
Comment (bharathguvvala) [inline pkg/reco/workflow.go:94]: Add a comment as a reminder to log an event/metric about the recommendation config generated by the recommender.

File: pkg/reco/workflow.go [hunk 2]
```diff
@@ -168,3 +169,13 @@ func pickSafestPolicy(p1, p2 *Policy) *Policy {
 		return p2
 	}
 }
+
+func transformTargetRecoConfig(targetRecoConfig *v1alpha1.HPAConfiguration) *v1alpha1.HPAConfiguration {
+	maxReplicas := targetRecoConfig.Max
+	minReplicas := targetRecoConfig.Min
+	optimalTargetUtil := targetRecoConfig.TargetMetricValue
```
Comment (bharathguvvala) [inline pkg/reco/workflow.go:176]: You can avoid using this additional local var. Instead use this directly in the return value.

File: pkg/reco/workflow.go [hunk 3]
```diff
@@ -168,3 +169,13 @@ func pickSafestPolicy(p1, p2 *Policy) *Policy {
 		return p2
 	}
 }
+
+func transformTargetRecoConfig(targetRecoConfig *v1alpha1.HPAConfiguration) *v1alpha1.HPAConfiguration {
+	maxReplicas := targetRecoConfig.Max
```
Comment (bharathguvvala) [inline pkg/reco/workflow.go:174]: add a nil check of the input

File: pkg/reco/workflow.go [hunk 4]
```diff
@@ -168,3 +170,15 @@ func pickSafestPolicy(p1, p2 *Policy) *Policy {
...
 	}
 }
+
+func transformTargetRecoConfig(targetRecoConfig *v1alpha1.HPAConfiguration) *v1alpha1.HPAConfiguration {
+	if targetRecoConfig == nil {
+		return nil
+	}
+	maxReplicas := targetRecoConfig.Max
+	minReplicas := targetRecoConfig.Min
+	if maxReplicas >= 3 && minReplicas < 3 {
```
Comment (bharathguvvala) [inline pkg/reco/workflow.go:180]: can we make this config driven instead of hardcoding?

File: pkg/reco/workflow.go (hunk 2 above)
Comment (deefreak) [reply pkg/reco/workflow.go]: sure

File: pkg/reco/workflow.go (hunk 4 above)
Comment (deefreak) [reply pkg/reco/workflow.go:180]: ok, naming it minRequiredReplicas in the config?

Comment (bharathguvvala) [reply pkg/reco/workflow.go:180]: sure

**Review by bharathguvvala — APPROVED** [review]

**Pull request merged**